core/src/test/java/org/apache/metamodel/query/FromItemTest.java

```diff
@@ -98,4 +105,35 @@ public void testSubQueryJoinToString() throws Exception {
 				"MetaModelSchema.project a LEFT JOIN (SELECT c.contributor_id, c.project_id AS foobar, c.name FROM MetaModelSchema.role c) b ON a.project_id = b.foobar",
 				from.toString());
 	}
+	
+    public void testCompoundJoin() {
+        final Schema schema = getExampleSchema();
+   
+        final QueryPostprocessDataContext dc = new QueryPostprocessDataContext() {
+            @Override
+            protected DataSet materializeMainSchemaTable(Table table, Column[] columns, int maxRows) {
+                Object[] values = new Object[columns.length];
+                for (int i = 0; i < columns.length; i++) {
+                    values[i] = columns[i].getColumnNumber();
+                }
+                DataSetHeader header = new SimpleDataSetHeader(columns);
+                DefaultRow row = new DefaultRow(header, values);
+                return new InMemoryDataSet(row);
+            }
+
+            @Override
+            protected String getMainSchemaName() throws MetaModelException {
+                return "MetaModelSchema";
+            }
+
+            @Override
+            protected Schema getMainSchema() throws MetaModelException {
+                return schema;
+            }
+        };
+ 
+        Query query = dc.parseQuery("SELECT c.contributor_id,p.project_id from contributor c INNER JOIN role r ON c.contributor_id=r.contributor_id INNER JOIN project p ON p.project_id=r.project_id");
+        System.out.println(query);
```

Comment:
Probably a left-over from a debugging session or so, but let's not print to syso.